Added a check inside WP_Block_Styles_Registry::register throw space error

amustaque97 opened the pull request:
<!--
Hi there! Thanks for contributing to WordPress!

Pull Requests in this GitHub repository **must** be linked to a ticket in the WordPress Core Trac instance (https://core.trac.wordpress.org), and are only used for code review. **No pull requests will be merged on GitHub.**

See the WordPress Handbook page on using PRs for Code Review more information: https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/

If this is your first time contributing, you may also find reviewing these guides first to be helpful:
- FAQs for New Contributors: https://make.wordpress.org/core/handbook/tutorials/faq-for-new-contributors/
- Contributing with Code Guide: https://make.wordpress.org/core/handbook/contribute/
- WordPress Coding Standards: https://make.wordpress.org/core/handbook/best-practices/coding-standards/
- Inline Documentation Standards: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/
- Browser Support Policies: https://make.wordpress.org/core/handbook/best-practices/browser-support/
- Proper spelling and grammar related best practices: https://make.wordpress.org/core/handbook/best-practices/spelling/
-->

<!-- Insert a description of your changes here -->

Trac ticket: <!-- insert a link to the WordPress Trac ticket here -->
https://core.trac.wordpress.org/ticket/54296

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

amustaque97 committed:
Added a check inside WP_Block_Styles_Registry::register throw space error

Fixes #54296 originally it was reported in the gutenberg repo issue #34242
In this commit, I have added a check inside the function if block style name
contain space it will trigger a notice and return false.

I have added a test case as well for this scenario

amustaque97 committed:
Merge branch 'master' into fix/block-style-name-error-space

amustaque97 committed:
fixing lint errors

Commit:
Replaced preg_match with the strpos function call.

Since we  just need to check for any single space which can be
checked easily. preg_match I fill is a bit overkill for this.

afragen commented on src/wp-includes/class-wp-block-styles-registry.php:
```diff
@@ -60,6 +60,15 @@ public function register( $block_name, $style_properties ) {
 			return false;
 		}
 
+		if ( strpos( $style_properties['name'], ' ' ) !== false ) {
```
Reverse this conditional you should. 
— Yoda

afragen replied:
Was a review much earlier. I guess that has lees visibility.

amustaque97 replied:
It is already addressed. Marking it as resolve.

costdev reviewed (changes requested):
This review suggests some changes for consistency in documentation and the use of the `@covers` annotation for PHPUnit to improve coverage report accuracy.

amustaque97 committed:
Update tests/phpunit/tests/blocks/register.php

Following consistency in the documentation. Period should be added at the end of comment.

Co-authored-by: Colin Stewart <79332690+costdev@users.noreply.github.com>

amustaque97 committed:
Added @covers for code coverage

Co-authored-by: Colin Stewart <79332690+costdev@users.noreply.github.com>

costdev reviewed (approved):
Requested changes have been committed and CI passes. This review is complete.

hellofromtonya commented on src/wp-includes/class-wp-block-styles-registry.php:
```diff
@@ -60,6 +60,15 @@ public function register( $block_name, $style_properties ) {
 			return false;
 		}
 
+		if ( strpos( $style_properties['name'], ' ' ) !== false ) {
```
This could be `str_contains` instead to make it more readable:

```suggestion
		if ( str_contains( $style_properties['name'], ' ' ) ) {
```

Core now has a polyfill for using `str_contains()` https://www.php.net/manual/en/function.str-contains.php.

pbiron commented on src/wp-includes/class-wp-block-styles-registry.php:
```diff
@@ -60,6 +60,15 @@ public function register( $block_name, $style_properties ) {
 			return false;
 		}
 
+		if ( strpos( $style_properties['name'], ' ' ) !== false ) {
```
Shouldn't this should use a Yoda condition?, e.g.,

`if ( false !== strpos( $style_properties['name'], ' ' ) ) {}`

SergeyBiryukov replied:
Yeah, either that or using `str_contains()` as suggested above would be my preference too.

pbiron replied:
agreed...`str_contains()` is better and then no need for Yoda

pbiron reviewed (changes requested):
Suggestion to use a Yoda condition.

hellofromtonya committed:
Use str_contains() for readability.

amustaque97 committed:
Merge branch 'master' into fix/block-style-name-error-space

amustaque97 committed:
Merge branch 'trunk' into fix/block-style-name-error-space

amustaque97 committed:
Merge remote-tracking branch 'origin/trunk' into fix/block-style-name-error-space

hellofromtonya committed:
Adds additional test datasets.

SergeyBiryukov commented on tests/phpunit/tests/blocks/register.php:
```diff
@@ -559,4 +559,61 @@ public function test_filter_block_registration_metadata_settings() {
 	}
+
+	/**
+	 * Test case to validate `__doing_it_wrong()` when block style name attribute
```
`_doing_it_wrong()` starts with one underscore, not two :)

SergeyBiryukov commented on tests/phpunit/tests/blocks/register.php:
```diff
@@ -559,4 +559,61 @@ public function test_filter_block_registration_metadata_settings() {
+	}
+
+	/**
+	 * Test case to validate no `__doing_it_wrong()` happens when there's
```
Same here.

hellofromtonya committed:
Correct `_doing_it_wrong()` in test docblock.

hellofromtonya reviewed (approved):
Great job everyone 👏 This is ready for `commit` 👍

audrasjb commented:
Committed in https://core.trac.wordpress.org/changeset/52261
